x/storage/keeper/virtualgroup.go

```diff
@@ -122,6 +123,9 @@ func (k Keeper) VerifyGVGSecondarySPsBlsSignature(ctx sdk.Context, gvg *vgtypes.
 		if !found {
 			panic("should not happen")
 		}
+		if secondarySp.Status != sptypes.STATUS_IN_SERVICE {
```

Comment:
I think no need for this, as the SP already sign the message, so it must ensure store the data properly.

Reply:
removed